pkg/operation/botanist/addons.go

```diff
@@ -359,6 +362,65 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 		}
 	}
 
+	var (
+		poolNameVersionToImage = make(workerPoolKubeProxyImages)
+		kubernetesVersion      = b.Shoot.GetInfo().Spec.Kubernetes.Version
+	)
+
+	for _, worker := range b.Shoot.GetInfo().Spec.Provider.Workers {
+		image, err := b.ImageVector.FindImage(charts.ImageNameKubeProxy, imagevector.RuntimeVersion(kubernetesVersion), imagevector.TargetVersion(kubernetesVersion))
+		if err != nil {
+			return nil, err
+		}
+
+		poolNameVersionToImage.Upsert(worker.Name, kubernetesVersion, image.String())
+	}
+
+	nodeList := &corev1.NodeList{}
+	if err := b.K8sShootClient.Client().List(ctx, nodeList); err != nil {
+		return nil, err
+	}
+
+	for _, node := range nodeList.Items {
+		poolName, ok1 := node.Labels[v1beta1constants.LabelWorkerPool]
+		kubernetesVersion, ok2 := node.Labels[v1beta1constants.LabelWorkerKubernetesVersion]
+		if !ok1 || !ok2 {
+			continue
+		}
+
+		image, err := b.ImageVector.FindImage(charts.ImageNameKubeProxy, imagevector.RuntimeVersion(kubernetesVersion), imagevector.TargetVersion(kubernetesVersion))
+		if err != nil {
+			return nil, err
+		}
+
+		poolNameVersionToImage.Upsert(poolName, kubernetesVersion, image.String())
+	}
+
+	var workerPools []map[string]string
+
+	// TODO(rfranzke): Delete this in a future version.
+	{
+		kubeProxyImage, err := b.ImageVector.FindImage(charts.ImageNameKubeProxy, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+		if err != nil {
+			return nil, err
+		}
+
+		workerPools = append(workerPools, map[string]string{
+			"name":              "",
+			"kubernetesVersion": b.Shoot.GetInfo().Spec.Kubernetes.Version,
+			"kubeProxyImage":    kubeProxyImage.String(),
+		})
+	}
+
+	for _, obj := range poolNameVersionToImage {
+		workerPools = append(workerPools, map[string]string{
+			"name":              obj.poolName,
+			"kubernetesVersion": obj.kubernetesVersion,
+			"kubeProxyImage":    obj.image,
+		})
+	}
+	kubeProxyConfig["workerPools"] = workerPools
```

Comment:
Do you really need this? Can't you do something like:
```suggestion
	var workerPools []workerPoolKubeProxyImage
	...
	for _, obj := range poolNameVersionToImage {
		workerPools = append(workerPools, obj)
	}
	kubeProxyConfig["workerPools"] = workerPools
```

Reply:
This is not possible since unexported struct attributes cannot be marshalled by JSON. I could export them and change the Helm chart values accordingly, but it doesn't yield any benefit, especially in the light of the planned work w.r.t. #2754 where all of this will be dropped/implement cleaner.

Reply:
Well, I would still avoid this "manual marshalling" even if it's temporary. But you are right that this is rather minor.